### PR TITLE
Use disyuntor

### PR DESCRIPTION
**Pull request by elbuo8**
:D

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -57,6 +57,12 @@ function BaaSClient (options, done) {
   }
 
   this._pendingRequests = 0;
+
+  this._sendRequestSafe = disyuntor(this._sendRequest.bind(this), Object.assign({
+    name: 'baas.client',
+    timeout: options.requestTimeout
```
Comment (dschenkelman) [inline client.js]: no error logging?

**Comment by dschenkelman** [comment]
would be interesting to add a couple of tests

File: client.js (hunk 1 above)
Comment (elbuo8) [inline client.js]: You can pass `options.monitor` with a function that can log/whatever.
https://github.com/auth0/disyuntor/blob/master/index.js#L64

**Comment by elbuo8** [comment]
Tests pass intermittently locally, always fail in travis. Will check what is up tomorrow.

**Comment by jfromaniello** [comment]
Instead of having an option.monitor, maybe we can emit an event since its already an eventemitter

**Comment by elbuo8** [comment]
@jfromaniello adding it.

**Pull request merged**